comid/cond_endorse_series_triple_test.go

```diff
@@ -4,44 +4,663 @@
 package comid
 
 import (
+	"encoding/json"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/veraison/corim/extensions"
 )
 
-func Test_CondEndorseSeries_NewCondEndorseSeriesTriples_OK(t *testing.T) {
+func Test_CondEndorseSeriesTriple_NewCondEndorseSeriesTriples_OK(t *testing.T) {
 	c := NewCondEndorseSeriesTriples()
 	require.NotNil(t, c)
+	assert.True(t, c.IsEmpty())
 }
 
-func Test_CondEndorseSeries_Valid_NOK(t *testing.T) {
+func Test_CondEndorseSeriesTriple_Add_Valid(t *testing.T) {
 	c := NewCondEndorseSeriesTriples()
-	expectedErr := "error at index 0: stateful environment validation failed: environment validation failed: environment must not be empty"
-	series := &CondEndorseSeriesTriple{}
+	// Create a valid triple
+	triple := &CondEndorseSeriesTriple{
+		Condition: ValueTriple{
+			Environment: Environment{
+				Class: &Class{
+					ClassID: MustNewUUIDClassID(TestUUID),
+				},
+			},
+		},
+		Series: *NewCondEndorseSeriesRecords(),
+	}
+
+	c.Add(triple)
+	assert.False(t, c.IsEmpty())
+	err := c.Valid()
+	assert.Contains(t, err.Error(), "no measurement entries")
+}
+
+func Test_CondEndorseSeriesTriple_Valid_EmptyCondition(t *testing.T) {
+	c := NewCondEndorseSeriesTriples()
+	series := &CondEndorseSeriesTriple{
```

Comment:
As you are using the word series for  `series` part of `series: [ + conditional-series-record ]`, I would request you use a variable name ces or something similar to avoid the confusion in all the tests